lib/kvdb/kvdb_home.c

```diff
@@ -36,7 +36,8 @@ path_copy(const char *home, const char *path, char *buf, const size_t buf_sz)
         return 0;
     }
 
-    n = snprintf(buf, buf_sz, "%s/%s", home, path);
+    n = snprintf(
+        buf, buf_sz, "%s%s%s", home, home[strnlen(home, PATH_MAX) - 1] == '/' ? "" : "/", path);
```

Comment:
If home is longer than PATH_MAX it'll look at the wrong character...  Maybe you should check home first, something like:

	n = strnlen(home, PATH_MAX);
	if (n >= PATH_MAX)
		return merr(ENAMETOOLONG);

	while (n > 0 && home[n - 1] == '/')
		--n;

Reply:
By this point in the code, the home has already been validated to be less than PATH_MAX. Happy to add some code to assert that though in lieu of returning an error that could never happen, if you agree. We could also do an strnlen() check in hse_kvdb_open()/create()/drop() which could be more appropriate as another form of validating user input instead of waiting potentially until pidfile creation to find out the path was too long.